http/servlet-undertow/src/test/java/io/quarkus/ts/http/undertow/HttpServletWithSessionListenerIT.java

```diff
@@ -51,6 +63,27 @@ public void sessionSecured() {
         thenCheckActiveSessionsEqualTo(1);
     }
 
+    @Test
+    @DisplayName("Undertow servlet test http2")
+    public void underTowHttp2Servlet(VertxTestContext vertxTestContext) {
+        Vertx vertx = Vertx.vertx();
+        HttpClientOptions options = new HttpClientOptions().setProtocolVersion(HttpVersion.HTTP_2);
+        HttpClient httpClient = vertx.createHttpClient(options);
+        httpClient.request(HttpMethod.GET, 1101, "localhost", "/app/servlet/hello")
+                .compose(request -> request.send()
+                        .compose(httpClientResponse -> {
+                            assertEquals(HttpStatus.SC_OK, httpClientResponse.statusCode());
+                            assertEquals(HttpVersion.HTTP_2, httpClientResponse.version());
+                            return httpClientResponse.body();
+                        }))
+                .onSuccess(body -> vertxTestContext.verify(() -> {
+                    assertThat("Body response", body.toString().contains("Hello World"));
+                    vertxTestContext.completeNow();
+                    httpClient.close();
```

Comment:
so you are not going to close the `httpClient` and `vertx` on failure? We are re-running failed tests, are you sure this won't leave Vert.x unclosed during other tests? I suppose when:

- on success (done)
- on failure
- on timeout - by which I mean that you call vertxTestContext.completeNow, therefore I presume if yo never call it, test will eventually timeout. If you never receive response from backend, it will depend whether test timeout is shorter than Http client timeout. What is the default client timeout? I don't expect any so we need to take into consideration test timeout.

I don't think it is very important, but we can do it right because I think it shouldn't be hard. Let me know if you disagree.

Reply:
if unit test times out, aftereach method is called; I'll close this, let's keep it.